src/charm.py

```diff
@@ -110,335 +202,226 @@ def set_pod_spec(self, event):
+            raise e
+        self.unit.status = MaintenanceStatus("K8S resources removed")
 
-        self.msg = str(msg)
-        self.status_type = status_type
-        self.status = status_type(self.msg)
+    def _gen_certs(self):
+        """Generate certificates."""
+        # generate SSL configuration based on template
+        model = self.model.name
+
+        try:
+            ssl_conf_template = open(SSL_CONFIG_FILE)
+            ssl_conf = ssl_conf_template.read()
+        except ApiError as error:
+            self.logger.warning(f"Failed to open SSL config file: {error}")
+
+        ssl_conf = ssl_conf.replace("{{ model }}", str(model))
+        Path("/tmp/seldon-cert-gen-ssl.conf").write_text(ssl_conf)
+
+        # execute OpenSSL commands
+        check_call(["openssl", "genrsa", "-out", "/tmp/seldon-cert-gen-ca.key", "2048"])
+        check_call(["openssl", "genrsa", "-out", "/tmp/seldon-cert-gen-server.key", "2048"])
+        check_call(
+            [
+                "openssl",
+                "req",
+                "-x509",
+                "-new",
+                "-sha256",
+                "-nodes",
+                "-days",
+                "3650",
+                "-key",
+                "/tmp/seldon-cert-gen-ca.key",
+                "-subj",
+                "/CN=127.0.0.1",
+                "-out",
+                "/tmp/seldon-cert-gen-ca.crt",
+            ]
+        )
+        check_call(
+            [
+                "openssl",
+                "req",
+                "-new",
+                "-sha256",
+                "-key",
+                "/tmp/seldon-cert-gen-server.key",
+                "-out",
+                "/tmp/seldon-cert-gen-server.csr",
+                "-config",
+                "/tmp/seldon-cert-gen-ssl.conf",
+            ]
+        )
+        check_call(
+            [
+                "openssl",
+                "x509",
+                "-req",
+                "-sha256",
+                "-in",
+                "/tmp/seldon-cert-gen-server.csr",
+                "-CA",
+                "/tmp/seldon-cert-gen-ca.crt",
+                "-CAkey",
+                "/tmp/seldon-cert-gen-ca.key",
+                "-CAcreateserial",
+                "-out",
+                "/tmp/seldon-cert-gen-cert.pem",
+                "-days",
+                "365",
+                "-extensions",
+                "v3_ext",
+                "-extfile",
+                "/tmp/seldon-cert-gen-ssl.conf",
+            ]
+        )
+
+        ret_certs = {
+            "cert": Path("/tmp/seldon-cert-gen-cert.pem").read_text(),
+            "key": Path("/tmp/seldon-cert-gen-server.key").read_text(),
+            "ca": Path("/tmp/seldon-cert-gen-ca.crt").read_text(),
+        }
+
+        # cleanup temporary files
+        check_call(["rm", "-f", "/tmp/seldon-cert-gen-*"])
+
+        return ret_certs
+
+    #
+    # Main entry point for the Charm
+    #
```

Comment:
I think this becomes redundant now with the docstring

Reply:
Yeah, missed it.

Reply:
Done.